File: src/main/java/kr/co/lunatalk/domain/order/domain/OptionSnapshot.java
package kr.co.lunatalk.domain.order.domain;

import jakarta.persistence.Embeddable;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Embeddable
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
public class OptionSnapshot {
	private String color;

	@Builder(access = AccessLevel.PRIVATE)
	public OptionSnapshot(String color) {
		this.color = color;
	}


	public static OptionSnapshot createOptionSnapshot(String color) {
		return OptionSnapshot.builder()
			.color(color)
			.build();
	}

}

File: lunatalk-core/src/main/kotlin/kr/co/lunatalk/domain/order/domain/OptionSnapshot.kt
package kr.co.lunatalk.domain.order.domain

import jakarta.persistence.Embeddable

@Embeddable
open class OptionSnapshot protected constructor() {

    open var color: String? = null
        protected set

    private constructor(color: String?) : this() {
        this.color = color
    }

    companion object {
        fun createOptionSnapshot(color: String?): OptionSnapshot {
            return OptionSnapshot(color)
        }
    }
}
